modules/hitlnext/src/views/full/app/components/UserProfile.tsx

```diff
@@ -0,0 +1,84 @@
+import { Collapsible, lang } from 'botpress/shared'
+import _ from 'lodash'
+import React, { FC, Fragment, useContext, useEffect, useState } from 'react'
+
+import { IUser } from '../../../../types'
+import style from '../../style.scss'
+import { Context } from '../Store'
+
+import { generateUsername, getOrSet } from './../utils'
+
+const UserProfile: FC<IUser> = props => {
+  const { state, dispatch } = useContext(Context)
+
+  const [expanded, setExpanded] = useState(true)
+  const [defaultUsername, setDefaultUsername] = useState<string>()
+
+  useEffect(() => {
+    const key = props.id
+    const username = getOrSet(
+      () => {
+        return _.get(state, `defaults.user.${key}.username`)
+      },
+      value => {
+        dispatch({
+          type: 'setDefault',
+          payload: {
+            user: {
+              [key]: {
+                username: value
+              }
+            }
+          }
+        })
+      },
+      generateUsername()
+    )
+
+    setDefaultUsername(username)
```

Comment:
This exact method also exists elsewhere. Can you use the same everywhere? (no duplicates)